notebooks/Lesson-1-Models-Prompts-and-Output-Parsers.ipynb

```diff
@@ -0,0 +1,1066 @@
+    "cleaning up me kitchen. I need yer help \\\n",
+    "right now, matey!\n",
+    "\"\"\"\n",
+    "\n",
+    "# Style of response\n",
+    "style = \"\"\"American English \\\n",
+    "in a calm and respectful tone\n",
+    "\"\"\"\n",
+    "\n",
+    "# Prompt for the model\n",
+    "prompt = f\"\"\"Translate the text \\\n",
+    "that is delimited by triple backticks \n",
+    "into a style that is {style}.\n",
+    "text: ```{customer_email}```\n",
+    "\"\"\"\n",
+    "\n",
+    "print(prompt)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"I'm upset that my blender lid flew off and splattered my kitchen walls with smoothie! And to make matters worse, the warranty doesn't cover the cost of cleaning up my kitchen. I need your help right now, friend.\""
+      ]
+     },
+     "execution_count": 45,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Use the translated prompt to generate a response\n",
+    "response = get_completion(prompt)\n",
+    "response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Chat API : LangChain"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before we begin working with Python, let's make sure we have the latest `langchain` installed from the command-line of our Python virtual environment:\n",
+    "\n",
+    "```sh\n",
+    "(.venv) % pip install --upgrade langchain\n",
+    "```\n",
+    "\n",
+    "Save the newly installed dependencies to `requirements.txt`:\n",
+    "\n",
+    "```sh\n",
+    "(.venv) % pip freeze > requirements.txt\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/rob/repos/deeplearning-ai/langchain-for-llm-application-development/.venv/lib/python3.11/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The class `langchain_community.chat_models.openai.ChatOpenAI` was deprecated in langchain-community 0.0.10 and will be removed in 0.2.0. An updated version of the class exists in the langchain-openai package and should be used instead. To use it run `pip install -U langchain-openai` and import as `from langchain_openai import ChatOpenAI`.\n",
+      "  warn_deprecated(\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "ChatOpenAI(client=<openai.resources.chat.completions.Completions object at 0x107ff3290>, async_client=<openai.resources.chat.completions.AsyncCompletions object at 0x111654ed0>, model_name='gpt-3.5-turbo-1106', temperature=0.0, openai_api_key='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', openai_proxy='')"
```

Comment:
Fah'k. This was just a demo OpenAI key, and immediately, I received a notice from OpenAI that this had been disabled. 😅

Shame on me!